website/src/docs/hotchocolate/defining-a-schema/relay.md

```diff
@@ -0,0 +1,540 @@
+{
+    public async Task<Product> GetProductAsync(string id)
+    {
+        // Omitted code for brevity
+    }
+}
+```
+
+</ExampleTabs.Code>
+<ExampleTabs.Schema>
+
+The approach of either Annotation-based or Code-first can be used in conjunction with Schema-first.
+
+</ExampleTabs.Schema>
+</ExampleTabs>
+
+Since node resolvers resolve entities by their Id, they are the perfect place to start utilizing DataLoaders.
+
+[Learn more about DataLoaders](/docs/hotchocolate/fetching-data/dataloader)
+
+# Connections
+
+_Connections_ are a standardized way to expose pagination capabilities.
+
+```sdl
+type Query {
+  users(first: Int after: String last: Int before: String): UserConnection
+}
+
+type UserConnection {
+  pageInfo: PageInfo!
+  edges: [UserEdge!]
+  nodes: [User!]
+}
+
+type UserEdge {
+  cursor: String!
+  node: User!
+}
+
+type PageInfo {
+  hasNextPage: Boolean!
+  hasPreviousPage: Boolean!
+  startCursor: String
+  endCursor: String
+}
+```
+
+[Learn more about Connections](/docs/hotchocolate/fetching-data/pagination#connections)
+
+# Query field in Mutation payloads
+
+It's a common best practice to return a payload type from mutations containing the affected entity as a field.
+
+```sdl
+type Mutation {
+  likePost(id: ID!): LikePostPayload
+}
+
+type LikePostPayload {
+  post: Post
+}
+```
+
+This allows us to immediately process the affected entity in the client application responsible for the mutation.
+
+Sometimes a mutation might also affect other parts of our application as well. Maybe the `likePost` mutation needs to update an Activity Feed.
+
+For this scenario we can expose a `query` field on our payload type to allow the client application to fetch everything it needs to update its state in one round trip.
+
+```sdl
+type LikePostPayload {
+  post: Post
+  query: Query
+}
+```
+
+A resulting mutation request could look like the following.
+
+```graphql
+mutation {
+  likePost(id: 1) {
+    post {
+      id
+      content
+      likes
+    }
+    query {
+      ...ActivityFeed_Fragment
+    }
+  }
+}
+```
+
+## Usage
+
+Hot Chocolate allows us to automatically add this `query` field to all of our mutation payload types:
+
+```csharp
+public class Startup
+{
+    public void ConfigureServices(IServiceCollection services)
+    {
+        services
+            .AddGraphQLServer()
+            .AddQueryFieldToMutationPayloads();
+    }
+}
+```
+
+By default, this will add a field of type `Query` called `query` to each top-level mutation field type, whose name ends in `Payload`.
+
+Of course these defaults can be tweaked:
+
+```csharp
+services
+    .AddGraphQLServer()
+    .AddQueryFieldToMutationPayloads(options =>
+    {
+        options.QueryFieldName = "rootQuery";
+        options.MutationPayloadPredicate =
+            (type) => type.Name.Value.EndsWith("Result");
+    });
+```
+
+This would add a field of type `Query` with the name of `rootQuery` to each top-level mutation field type, whose name ends in `Result`.
```

Comment:
I think we need to add a note here that this query field doesn't work in stitching at the moment. 

```suggestion
This would add a field of type `Query` with the name of `rootQuery` to each top-level mutation field type, whose name ends in `Result`.

> Note: This feature currently doesn't work on a stitching gateway, however this will be addressed in a future release focused on stitching. It's tracked as [#3158](https://github.com/ChilliCream/hotchocolate/issues/3158).
```